### PR TITLE
Generic FileTable

### PR DESCRIPTION
**Pull request by StuartHarris**
Make the `FileTable` component generic, so that it can also be used to list workspace members.

### CODE REVIEW

File: frontend/components/Table/MobileList.test.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  folder: "fake-folder",
+  latestVersion: "v2",
+  modifiedAt: "2020-10-08T07:10:09Z",
+  title: "My Fake File",
+};
+
+describe(MobileList, () => {
+  it("renders title with a surrounding <a> when given a function component with a link", () => {
+    const { getByText } = render(
+      <MobileList
```
Comment (alsuren) [inline frontend/components/Table/MobileList.test.tsx:22]: add a test that uses icon here and in Table

File: frontend/components/Table/Table.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  folder: "fake-folder",
+  latestVersion: "v2",
+  modifiedAt: "2020-10-08T07:10:09Z",
+  title: "My Fake File",
+};
+
+describe(Table, () => {
+  it("renders title with a surrounding <a> when given a function component with a link", () => {
+    const { getByText } = render(
+      <Table
```
Comment (alsuren) [inline frontend/components/Table/Table.test.tsx:22]: also use icon in a test

File: frontend/components/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+              <NHSTable.Cell key={i}>{c.name}</NHSTable.Cell>
+            ))}
+          </NHSTable.Row>
+        </NHSTable.Head>
+        <NHSTable.Body>
+          {data.map((x) => (
+            <NHSTable.Row key={x.id}>
+              {columns.map(({ content }, i) => (
+                <NHSTable.Cell key={i}>
+                  {i == 0 && icon ? (
```
Comment (alsuren) [inline frontend/components/Table/Table.tsx:73]: ===

File: frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/index.tsx [hunk 4]
```diff
@@ -127,15 +165,23 @@ const FileHomepage = () => {
               "Loading..."
             ) : (
               <>
-                <MobileFileList
-                  files={[file.data.file]}
-                  workspaceId={workspaceId}
-                  titleLink={false}
+                <MobileList
+                  columns={[
+                    { content: IconCell },
```
Comment (alsuren) [inline frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/index.tsx:170]: extract icon here too

File: frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/index.tsx [hunk 5]
```diff
@@ -127,15 +165,23 @@ const FileHomepage = () => {
...
+                  data={[file.data.file as File]}
                 />
-                <FileTable
-                  files={[file.data.file]}
-                  workspaceId={workspaceId}
-                  titleLink={false}
+                <Table
+                  columns={[
+                    { name: "Title", content: IconCell },
+                    { content: TitleCell },
```
Comment (alsuren) [inline frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/index.tsx:180]: and here

File: frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/update-file.tsx [hunk 6]
```diff
@@ -89,20 +112,28 @@ const UpdateFile: NextPage<any> = ({ urqlClient }: { urqlClient: Client }) => {
...
             {file.data && (
-              <FileTable
+              <Table
                 tableHeading="Current file"
-                files={[file.data?.file]}
-                workspaceId={workspaceId}
-                titleLink={false}
+                columns={[
+                  { name: "Title", content: IconCell },
+                  { content: TitleCell },
```
Comment (alsuren) [inline frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/update-file.tsx:119]: and here

File: frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/update-file.tsx [hunk 7]
```diff
@@ -89,20 +112,28 @@ const UpdateFile: NextPage<any> = ({ urqlClient }: { urqlClient: Client }) => {
...
 
             {file.data && (
-              <MobileFileList
+              <MobileList
                 tableHeading="Current file"
-                files={[file.data?.file]}
-                workspaceId={workspaceId}
-                titleLink={false}
+                columns={[
+                  { content: IconCell },
```
Comment (alsuren) [inline frontend/pages/workspaces/[workspaceId]/folders/[folderId]/files/[fileId]/update-file.tsx:131]: and here

**Review by alsuren — APPROVED** [review]
@declanslevin biggest risk in this refactor is that we changed how the desktop table renders icons. They are no longer in their own cell, and we set the padding-right by hand.

**Comment by declanslevin** [comment]
Happy for this to be merged once the build is green 👍

**Pull request merged**